### DISCUSSION TITLE
Is overriding node labels with `__label__` considered safe?

### DISCUSSION THREAD

**Discussion by nquinn-ek** [neo4j-contrib/neomodel#711] [Q&A] [answered]
Hi,

I wanted to name my Neomodel model classes with a trailing "Node", for instance `PersonNode`, in order to distinguish them from other types of models (specifically, Pydantic models). However, I don't want all my nodes in the graph to have "Node" in the name, so I looked in the docs for a way to override the label instead of inheriting it from the class, and found nothing. 

But, from looking at the `__new__` method for the `NodeMeta` class, it appears that the label of the node is stored in the `__label__` property, and that label is taken from the name of the class unless already set:
```python
cls.__label__ = namespace.get("__label__", name)
```
So it appears we can override the label of the node by setting this as a attribute on the class:
```python
class PersonNode(neomodel.StructuredNode):
    __label__ = "Person"
```
I tested it in the CLI and it seems to work. It seems similar to what you can do in [SQLAlchemy](https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#declarative-table-with-mapped-column) to point to the name of the table:
```python
class User(Base):
    __tablename__ = "user"
```

So I am wondering, does anyone know if this considered part of the public API and safe to hook into, or a private implementation detail that it's risky to rely on?

**Comment by aanastasiou** [comment, marked as answer]
@nquinn-ek I would not call it "best practice" (`__label__` is "private" anyway), as it would differentiate what you perceive as an entity name in your program VS the backend. The label of a given entity is very important for the proper serialisation / deserialisation of the entity with the backend as well as setting up any indices. Most of the times, you don't have to deal with setting it manually.

Having said that, if you are extending a data model and you have such clashes, you could resolve them by modifying the `__label__` as a last resort. It would be risky if you run across situations where you have to re-set that label a few times, in which case you would have to cleanup any remnants manually.

Hope this helps.

**Reply by nquinn-ek** [reply]
thanks!

**Comment by yachal-upson** [comment]
Here for the same reason. Frustrated that __label__ is inferred from class name. 
In my case I'm developing a FastAPI backend so have Pydantic models sitting alongside Neomodel models, with a need to differentiate for sanity. Like the OP said, I too wanted to append "NodeSchema" to my neomodels, and "NodeModel" to my pydantics. 

Had to resort to setting __label__ at declaration (will not be touched after). Would have almost been nice to have a public class attribute.

**Reply by nquinn-ek** [reply]
looks like per the [latest docs](https://neomodel.readthedocs.io/en/latest/extending.html#custom-label), this is now considered to be supported. Also @yachal-upson that's my use case too - probably a pretty common one these days!
